codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentBuilderGenerator.kt

```diff
@@ -150,8 +170,85 @@ class FluentBuilderGenerator(
                 }
                 """,
                 *scope,
+                "epilogue" to (
+                    handleEventStreamInitialResponse() ?: writable {
+                        rustTemplate("#{Operation}::orchestrate(&runtime_plugins, input).await", *scope)
+                    }
+                ),
+            )
+        }
+
+    private fun handleEventStreamInitialResponse(): Writable? {
+        if (codegenContext.protocolImpl?.httpBindingResolver?.handlesEventStreamInitialResponse(operation) != true) {
+            return null
+        }
+
+        // The variable eventStreamMember is guaranteed to be non-null because of the short-circuit above
+        val eventStreamMember =
+            outputShape.members().find { member ->
+                member.isEventStream(codegenContext.model)
+            }!!
+
+        return writable {
+            val eventStreamMemberName = symbolProvider.toMemberName(eventStreamMember)
+            rustTemplate(
+                """
+                let mut output =
+                    #{Operation}::orchestrate(
+                        &runtime_plugins,
+                        input,
+                    )
+                    .await?;
+
+                // Converts any error encountered beyond this point into an `SdkError` response error
```

Comment:
What happens with non-streaming errors? Do we retain and expose the headers in those cases?

Reply:
In the case of non-streaming errors, the `?` operator at line 201 will return them just as usual. Anything after line 203 is because of streaming-related (more specifically handling initial-response-related) errors.